x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/slm/SnapshotInvocationRecord.java

```diff
@@ -86,6 +103,9 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         builder.startObject();
         {
             builder.field(SNAPSHOT_NAME.getPreferredName(), snapshotName);
+            if (builder.getRestApiVersion().matches(START_TIMESTAMP.getForRestApiVersion())) {
```

Comment:
Same here for the REST compatibility things, we can do
```suggestion
            if (snapshotStartTimestamp != null) {
```

Instead

Reply:
So since snapshotStartTimestamp is nullable, I assume we ought to change it to a Long instead of a long. The other field (currently called timestamp) will remain a long since it's not nullable.